### PR TITLE
Merge OngXinBin Comprehensive testing

### PR DESCRIPTION
**Pull request by n3wsoldier**
Add
- JUnit tests for Bookmark and its parsers

Update
- AnimeData and Bookmark remove unused method
- Checkstyle fixes

### CODE REVIEW

File: src/main/java/anichan/parser/BookmarkParser.java [hunk 1]
```diff
@@ -80,10 +82,11 @@ private void parameterParser(String paramGiven) throws AniException {
             break;
         case LIST_PARAM:
```
Comment (ChanJianHao) [inline src/main/java/anichan/parser/BookmarkParser.java:83]: Perhaps you could standardize params naming to be `PARAMETER_NAME`?

File: src/test/java/anichan/parser/BookmarkParserTest.java [hunk 2]
```diff
@@ -101,8 +180,28 @@ void execute_invalidFieldForEdit_ThrowsAniException() {
...
+    }
+
+    @Test
+    void execute_invalidFieldForNote_ThrowsAniException() {
+        BookmarkParser test1 = new BookmarkParser();
+        assertThrows(AniException.class, () -> {
+            test1.parse(INVALID_FIELD_NOTE_TEST1);
+        });
+
+        BookmarkParser test2 = new BookmarkParser();
```
Comment (ChanJianHao) [inline src/test/java/anichan/parser/BookmarkParserTest.java:207]: Great work in writing tests!

File: src/test/java/anichan/anime/AnimeDataTest.java [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
-
+    private static AnimeData animeData;
+    private static final Integer NEGATIVE_INTEGER = -1;
+    private static final Integer NULL_PARAM = null;
+    private static final String FIRST_INDEX_ANIME_INFO = "Index: 1" + System.lineSeparator()
+            + "Name: Cowboy Bebop" + System.lineSeparator()
+            + "Episodes: 26" + System.lineSeparator()
+            + "Release Date: 03/Apr/1998" + System.lineSeparator()
+            + "Rating: 86" + System.lineSeparator()
+            + "Genre: [Action, Adventure, Drama, Sci-Fi]";
```
Comment (ChanJianHao) [inline src/test/java/anichan/anime/AnimeDataTest.java:19]: Great work separating long strings into multi lines.

File: src/test/java/anichan/command/BookmarkAnimeCommandTest.java [hunk 4]
```diff
@@ -1,9 +1,215 @@
...
+        bookmark = activeWorkspace.getBookmark();
+    }
+
+    @Test
+    @Order(1)
+    void execute_validAddAnimeIndex_Successful() throws AniException {
+        BookmarkParser testParse = new BookmarkParser();
+        BookmarkAnimeCommand testBookmarkCommand = testParse.parse(VALID_ADD_TEST);
+        String result = testBookmarkCommand.execute(animeData, storageManager, user);
+        System.out.println(result);
```
Comment (ChanJianHao) [inline src/test/java/anichan/command/BookmarkAnimeCommandTest.java]: Is it good practice to print test results?

**Review by ChanJianHao — CHANGES_REQUESTED** [review]

File: src/main/java/anichan/parser/BookmarkParser.java (hunk 1 above)
Comment (n3wsoldier) [reply src/main/java/anichan/parser/BookmarkParser.java:83]: I understand the rationale with PARAMETER_NAME but I am consistent with the rest of our teammates,

**Review by ChanJianHao — APPROVED** [review]
LGTM!

**Review by EyoWeiChin — APPROVED** [review]
LGTM!! Thanks for the tests and refactoring

**Pull request merged**